src/util/RouterUtil.js

```diff
@@ -79,6 +79,18 @@ function (Okta, OAuth2Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
       return;
     }
 
+    if (err) {
+      if (!err.statusCode) {
+        // Bring the statusCode to the top-level of the Error
+        err.statusCode = err.xhr && err.xhr.status;
+      }
+      // Some controllers return the className as a function - process it here:
+      var className = typeof router.controller.className === 'function'
+        ? router.controller.className()
+        : router.controller.className;
+      router.controller.trigger('afterError', { controller: className }, err)
```

Comment:
how will this work for INVALID_TOKEN_EXCEPTION? we trigger this event, but right after they handle it, we navigate to a different page... maybe when INVALID_TOKEN_EXCEPTION is the error, we need to trigger `afterError` in the same code block where we handle `flashError`?
And then this logic can go below the special INVALID_TOKEN_EXCEPTION handler

Reply:
Yep, that is one approach - and I touched on that topic [here](https://github.com/okta/okta-signin-widget/pull/594#discussion_r237955733).

Users can enter an `INVALID_TOKEN_EXCEPTION ` state on any view/controller, where the proposed behavior is:

1. Trigger an `afterError` event on the controller the user is currently on
2. Set the current `flashError` state so we can use this later
    - Note: In the future, we can change how this works. By using the `afterError` event ourselves, we can handle this without needing to store the state and navigate.
3. Once navigating back to primary auth, trigger a new `error` event to build the banner with the expired token text.
   - Note: This message can be customized via `i18n` properties

To me, it seems that firing this event **where** it occurs is the right answer, as opposed to firing a new `afterError` event once we navigate back to primary auth.

What do you think?

Reply:
1. feels like we shall remove logic to line 103
2. and only trigger `afterError` for flashError when it's rendered. (after navigated back to primary)
3. since you already touch this `routeAfterAuthStatusChange`, why not let's extra the error handling logic into separated dedicated method

Reply:
I don't know if we need to remove 103, but I think this `if` block should go after handling INVALID_TOKEN_EXCEPTION... and that will go well with our description of `afterError`, since we will be triggering it AFTER the widget handled it (which in this case it means navigating to a different page)

Reply:
sorry I meant `moved` not `remove`..

Reply:
yeah, I think we are saying the same thing 👍 
Does that make sense @jmelberg-okta ?

Reply:
Got it @mauriciocastillosilva-okta and @haishengwu-okta! Good advice. I'll make that change now.

Reply:
Fixed with [`bfeae8d`](https://github.com/okta/okta-signin-widget/pull/594/commits/bfeae8dbf0950a79d3d5345e4e55ceb24d4c8f68)